### PR TITLE
fix: issue on generate template

### PR DESCRIPTION
**Pull request by juanpicado**
fix

```
➜ yo verdaccio-plugin

     _-----_     ╭──────────────────────────╮
    |       |    │        Welcome to        │
    |--(o)--|    │ generator-verdaccio-plug │
   `---------´   │   in plugin generator!   │
    ( _´U`_ )    ╰──────────────────────────╯
    /___A___\   /
     |  ~  |     
   __'.___.'__   
 ´   `  |° ´ Y ` 

? What is the name of your plugin? auth-keylock
? Select Language typescript
? What kind of plugin you want to create? middleware
? Please, describe your plugin saSasaSac d asdas d as dsa
? GitHub username or organization juanpicado
? Author's Name Juan Picado
? Author's Email xxxxxxx@gmail.com
? Key your keywords (comma to split) verdaccio,test,plugin,verdaccio-plugin
events.js:187
      throw er; // Unhandled 'error' event
      ^

AssertionError [ERR_ASSERTION]: Trying to copy from a source that does not exist: /Users/user/.nvm/versions/node/v13.1.0/lib/node_modules/generator-verdaccio-plugin/generators/app/templates/typescriptcommon/gitignore
```

add other features
- best template for middleware
- add verdaccio core libraries
- add jest by default
- custom `package.json` for each plugin: why? because each plugin has custom deps
- I added `REGISTRY_AUTH_TOKEN` to secrets
- Add workflow for automated release

next PR
- improve/add example template for storage
- improve/add example template for auth

### CODE REVIEW

File: generators/app/index.ts [hunk 1]
```diff
@@ -117,9 +116,10 @@ export default class VerdaccioPluginGenerator extends Generator {
   }
 
   packageJSON() {
-    const { lang } = this.props;
+    const { lang, pluginType } = this.props;
+    const pkgJsonLocation = `${lang}/${pluginType}/_package.json`;
     this.fs.copyTpl(
-      this.templatePath(lang + "/common/_package.json"),
```
Comment (anikethsaha) [inline generators/app/index.ts]: my bad

Comment (juanpicado) [reply generators/app/index.ts]: no problem, test should caught that.

Comment (anikethsaha) [reply generators/app/index.ts]: 👍

File: generators/app/templates/javascript/storage/_package.json [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  "description": "<%= description %>",
+  "main": "lib/index.js",
+  "types": "lib/index.d.ts",
+  "files": ["lib/"],
+  "engines": {
+    "node": ">=10"
+  },
+  "dependencies": {
+    "@verdaccio/commons-api": "^8.4.2",
+    "express": "^4.17.1"
```
Comment (anikethsaha) [inline generators/app/templates/javascript/storage/_package.json]: Why `express` in `storage` plugin ? is it needed ?

File: generators/app/templates/javascript/auth/_package.json [hunk 3]
```diff
@@ -8,13 +8,18 @@
   "engines": {
     "node": ">=10"
   },
+  "dependencies": {
+    "@verdaccio/commons-api": "^8.4.2",
+    "express": "^4.17.1"
+  },
   "devDependencies": {
-    "@babel/cli": "^7.7.0",
```
Comment (juanpicado) [inline generators/app/templates/javascript/auth/_package.json]: I remove this because `"@verdaccio/babel-preset` take care of it already.

Comment (anikethsaha) [reply generators/app/templates/javascript/auth/_package.json]: 👍 yay !!

File: generators/app/templates/javascript/common/_package.json [hunk 4]
```diff
@@ -7,19 +7,21 @@
     "node": ">=10"
   },
```
Comment (anikethsaha) [inline generators/app/templates/javascript/common/_package.json]: we are not using package.json from common right ? I guess you forgot to delete this file.

File: generators/app/templates/javascript/storage/_package.json (hunk 2 above)
Comment (juanpicado) [reply generators/app/templates/javascript/storage/_package.json]: My bad, i just copied the package Json, I will clean it up in the next PR

Comment (anikethsaha) [reply generators/app/templates/javascript/storage/_package.json]: cool 👍

**Comment by juanpicado** [comment]
Other templates still need work, but I don’t want to make it so big this PR

**Pull request merged**